spin/sessions.go

```diff
@@ -0,0 +1,35 @@
+package spin
+
+import (
+	uh "github.com/TRON-US/go-btfs/core/commands/storage/upload/helper"
+	"github.com/TRON-US/go-btfs/core/commands/storage/upload/sessions"
+	"github.com/TRON-US/go-btfs/core/commands/storage/upload/upload"
+
+	cmds "github.com/TRON-US/go-btfs-cmds"
+
+	"go4.org/syncutil"
+)
+
+func RenterSessions(req *cmds.Request, env cmds.Environment) {
+	params, err := uh.ExtractContextParams(req, env)
+	if err != nil {
+		return
+	}
+
+	cursor, err := sessions.GetRenterSessionsCursor(params)
+	if err != nil {
+		return
+	}
+
+	sem := syncutil.NewSem(10)
+	session, err := cursor.NextSession(sessions.RssWaitUploadReqSignedStatus)
+	for err == nil && session != nil {
```

Comment:
If no session on the first loop or some kind of error, this entire goroutine will exit ... then it will never work without restarting daemon.

Reply:
it works only when there were sessions during the previous daemon process.
new sessions will do the status transition with the state machine.